Source/JavaScriptCore/bytecode/PropertyCondition.cpp

```diff
@@ -224,7 +226,46 @@ bool PropertyCondition::isStillValidAssumingImpurePropertyWatchpoint(
         
         return true;
     }
-        
+
+    case AbsenceOfIndexed: {
+        if (structure->isDictionary()) {
+            if (PropertyConditionInternal::verbose)
+                dataLog("Invalid because it's a dictionary.\n");
+            return false;
+        }
```

Comment:
We still transition a dictionary if we add an indexing type, right?

Reply:
Ah, right, we can remove it.